tests/Tests.FeatureManagement/FeatureManagementTest.cs

```diff
@@ -1976,4 +1976,51 @@ public async Task CustomIFeatureDefinitionProvider()
             Assert.True(called);
         }
     }
+
+    public class PerformanceTests
```

Comment:
Can we use the tool like [benchmark .net](https://github.com/dotnet/BenchmarkDotNet) to verifiy this change can improve the performance?

Reply:
Can we create an issue to track this? It doesn't need to block v4 release.